lib/diagnostic.js

```diff
@@ -26,8 +26,30 @@
 const fs = require('fs'); // jshint ignore: line
 
 const sumLines = (filename, callback) => {
-  /* your response here */
-  callback();
+  fs.readFile(filename, 'utf8', (error, data) => {
+    if (error) {
+      console.error(error.stack);
+      return;
+    }
+    console.log(data);
+    let accum = 0;
+    let isError = false;
+    let lines = data.split('\n');
+    lines.pop();
+    lines.forEach( (line) => {
+      if(line === '') {
+        accum += 0;
+      } else if (isNaN(line)) {
+        isError = true;
+      } else {
+        accum += parseFloat(line);
+      }
+    });
+    if (isError) {
+      callback(new Error('line is not a number'));
```

Comment:
You can return your callback call here to prevent the call to `callback(error, accum)` from executing when `isError` is true and this will cause the final test to pass.